### PR TITLE
ENT-3332: Adds in visibility for vouchers and API endpoint to toggle

### PR DESCRIPTION
**Pull request by jlajoie**

### CODE REVIEW

File: ecommerce/extensions/api/serializers.py [hunk 1]
```diff
@@ -1262,6 +1269,15 @@ def get_sales_force_id(self, obj):
         except AttributeError:
             return None
 
+    def get_is_public_code(self, obj):
+        """
+        Get the Is Public Code boolean attached to the coupon.
+        """
+        try:
+            return obj.attr.is_publice_code
```
Comment (binodpant) [inline ecommerce/extensions/api/serializers.py]: is this a typo?

Comment (jlajoie) [reply ecommerce/extensions/api/serializers.py]: It is and that explains one of the test issues I have right now 😓

File: ecommerce/extensions/api/v2/tests/views/test_enterprise.py [hunk 2]
```diff
@@ -850,7 +850,7 @@ def test_coupon_codes_detail_csv(self):
         # Strip out first row (headers) and last row (extra csv line)
         csv_data = csv_content[1:-1]
         # Verify headers.
-        self.assertEqual(csv_header, 'assigned_to,assignment_date,code,last_reminder_date,redeem_url,'
+        self.assertEqual(csv_header, 'assigned_to,assignment_date,code,is_public,last_reminder_date,redeem_url,'
```
Comment (binodpant) [inline ecommerce/extensions/api/v2/tests/views/test_enterprise.py:853]: interested/curious about the ordering of the columns here.. should is_public be at the end? does it matter based on how the csv gen works?

Comment (jlajoie) [reply ecommerce/extensions/api/v2/tests/views/test_enterprise.py:853]: It's just alphabetical here

File: ecommerce/extensions/api/v2/views/enterprise.py [hunk 3]
```diff
@@ -676,6 +683,20 @@ def assign(self, request, pk):  # pylint: disable=unused-argument
...
+        """
+        Assign users by email to codes within the Coupon.
+        """
+        coupon = self.get_object()
+        codes = request.data.pop('code_ids')
+        is_public = request.data.pop('is_public')
+        if codes and is_public is not None:
+            coupon.attr.coupon_vouchers.vouchers.filter(code__in=codes).update(is_public=is_public)
+            return Response(status=status.HTTP_200_OK)
+        return Response(status=status.HTTP_400_BAD_REQUEST)
```
Comment (binodpant) [inline ecommerce/extensions/api/v2/views/enterprise.py]: optional: would it be wise to put some sort of message in the response to indicate what was the issue? such as 'code_ids' not provided, or is_public not provided? should be obvious to dev but better if we are going to any how surface these messages anywhere in the UIs later.. maybe

File: ecommerce/extensions/api/v2/views/enterprise.py [hunk 4]
```diff
@@ -378,6 +380,11 @@ def codes(self, request, pk, format=None):  # pylint: disable=unused-argument, r
         if not serializer_class:
             raise serializers.ValidationError('Invalid code_filter specified: {}'.format(code_filter))
 
+        if visibility_filter == VOUCHER_IS_PUBLIC:
+            queryset = queryset.filter(is_public=True)
+        elif visibility_filter == VOUCHER_IS_PRIVATE:
+            queryset = queryset.filter(is_public=False)
```
Comment (binodpant) [inline ecommerce/extensions/api/v2/views/enterprise.py:386]: what happens if someone sends in a visibility_filter that is neither of the two values?

Comment (binodpant) [reply ecommerce/extensions/api/v2/views/enterprise.py:386]: perhaps a validationError makes sense

**Review by binodpant — APPROVED** [review]
nothing major from me, the new field looks good to me, I did not try pulling this down and starting the ecom service and testing locally, but if that is desirable I can try that

**Pull request merged**